site/content/en/docs/concepts/cluster_queue.md

```diff
@@ -348,6 +350,37 @@ Workloads as possible, preferring Workloads with these characteristics:
 - Workloads with the lowest priority.
 - Workloads that have been admitted more recently.
 
+## FlavorFungibility
+
+When there is not enough nominal quota of resources in a ResourceFlavor, the incoming workload can borrow or preempt in ClusterQueue or Cohort.
+
+Kueue evaluates the flavors in a ClusterQueue in order. You can configure ClusterQueues to try preemption whenever preemption might help before trying next ResourceFlavor or only borrowing after all ResourceFlavors have been considered by setting `FlavorFungibility`.
+
+A configuration for a ClusterQueue that changes this behavior looks like the following:
+```yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ClusterQueue
+metadata:
+  name: "team-a-cq"
+spec:
+  flavorFungibility:
+    whenCanBorrow: TryNextFlavor
+    whenCanPreempt: Preempt
+```
+
+The fields above do the following:
+- `whenCanBorrow` determines whether a workload should stop finding a better assignment if it can get enough resource by borrowing in current ResourceFlavor. The possible values are:
+  - `Borrow` (default): ClusterQueue stops finding a better assignment.
+  - `TryNextFlavor`: ClusterQueue tries the next ResourceFlavor to see if the workload can get a better assignment.
+- `whenCanPreempt` determines whether a workload should try preemtion in current ResourceFlavor before try the next one. The possible values are:
+  - `Preempt`: ClusterQueue stops trying preemption in current ResourceFlavor and starts from the next one if preempting failed.
+  - `TryNextFlavor` (default): ClusterQueue tries the next ResourceFlavor to see if the workload can fit in the ResourceFlavor.
+
+By default, the incoming workload stops trying the next flavor if the workload can get enough borrowed resources. 
+And Kueue triggers preemption only after Kueue determines that the remaining ResourceFlavors can't fit the workload.
+
+Note that whenever possible and the configured policy allows, Kueue tries to avoid preemptions by borrowing. 
```

Comment:
```suggestion
Note that, whenever possible and when the configured policy allows it, Kueue avoids preemptions if it can fit a Workload by borrowing. 
```